packages/lesswrong/server/search/elastic/QueryParser.ts

```diff
@@ -0,0 +1,56 @@
+import trim from "lodash/trim";
+
+export type QueryToken = {
+  type: "should" | "must" | "not",
+  token: string,
+}
+
+export type QueryParserResult = {
+  tokens: QueryToken[],
+  isAdvanced: boolean,
+}
+
+class QueryParser {
```

Comment:
I think I feel the same way about this as I do about (ElasticSearchQuery)[https://github.com/ForumMagnum/ForumMagnum/pull/7188/files#r1197199219]